ClientApp/src/app/services/user.service.ts

```diff
@@ -16,19 +16,14 @@ export class UserService {
   ) {}
 
   public async getCurrentUser(): Promise<User> {
-    if (!this._user) {
-      try {
-        // First time or refresh
-        const token = this.sessionService.getToken();
-        const user = await this.userProvider.getUser(token);
-        this.setUser(user);
-        return this._user;
-      } catch (e) {
-        return null;
-      }
+    try {
+      const token = this.sessionService.getToken();
+      const user = await this.userProvider.getUser(token);
+      this.setUser(user);
+      return this._user;
+    } catch (e) {
+      return null;
```

Comment:
This should be its own function, maybe "refreshCurrentUser" or something.
That way "getCurrentUser" does not query the database when it does not need to 🔥